EventFilter/SiPixelRawToDigi/src/PixelDataFormatter.cc

```diff
@@ -204,6 +202,14 @@ void PixelDataFormatter::interpretRawData(
       local = std::make_unique<LocalPixel>(localDP);  // local pixel coordinate
     }
 
+    if (nrawId != rawId) {
+      nrawId = rawId;
+      detDigis = &digis.find_or_insert(rawId);
+      if ((*detDigis).empty()) {
+        (*detDigis).data.reserve(32);  // avoid the first relocations
+      }
+    }
+
     GlobalPixel global = rocp->toGlobal(*local);  // global pixel coordinate (in module)
     (*detDigis).data.emplace_back(global.row, global.col, adc);
     LogTrace("") << (*detDigis).data.back();
```

Comment:
Would a protection be needed here (see static analyzer report)?
```suggestion
    if (detDigis) {
      GlobalPixel global = rocp->toGlobal(*local);  // global pixel coordinate (in module)
      (*detDigis).data.emplace_back(global.row, global.col, adc);
      LogTrace("") << (*detDigis).data.back();
    }
```

Reply:
Thanks, @perrotta, for the suggestion. Indeed, this is good to add. Perhaps it would also be good to add a LogError as an else follow-up because a null pointer at this point is normally not expected.